### ISSUE TITLE
Trailing whitespace that could produce a hard break but won't due to context should also be flagged by MD009

### ISSUE THREAD

**Issue by TheJaredWilcurt** [DavidAnson/markdownlint#216]
```md
* Example
  * The next line has 2 spaces
  
  * The line above this should be flagged for trailing whitespace
```
There are two spaces on the 3rd line that should be flagged as trailing whitespace.
Here are the rules I'm using, in case I have something set wrong:

* https://github.com/TheJaredWilcurt/tjw-markdownlint-rules/blob/master/tjwmarkdownlint.yml

Here it is again, but with the spaces replaced by hearts to make it easier to see
```md
*♥asdf
♥♥*♥zxcv
♥♥
♥♥*♥qwer
```

**Comment by DavidAnson** [comment]
Exactly 2 trailing spaces has a special meaning in Markdown. If you change your example to just one space or three+, MD009 will fire as expected. You can use the `br_spaces` perimeter to report issues for two spaces as well. More detail is available  in the documentation here: https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md009---trailing-spaces
